internal/runbits/requirements/requirements.go

```diff
@@ -355,6 +423,85 @@ func (r *RequirementOperation) updateCommitID(commitID strfmt.UUID) error {
+		if err := r.Config.Set(constants.SecurityPromptLevelConfig, promptDefaultLevel); err != nil {
+			multilog.Error("Failed to set security prompt level config: %v", err)
+		}
+	}
+
+	prompt := r.Config.GetBool(constants.SecurityPromptConfig)
+	promptLevel := r.Config.GetString(constants.SecurityPromptLevelConfig)
+
+	switch promptLevel {
+	case vulnModel.SeverityCritical:
+		return prompt && len(vulnerabilities.Vulnerabilities.Critical) > 0
+	case vulnModel.SeverityHigh:
+		return prompt &&
+			(len(vulnerabilities.Vulnerabilities.Critical) > 0 ||
+				len(vulnerabilities.Vulnerabilities.High) > 0)
+	case vulnModel.SeverityMedium:
+		return prompt &&
+			(len(vulnerabilities.Vulnerabilities.Critical) > 0 ||
+				len(vulnerabilities.Vulnerabilities.High) > 0 ||
+				len(vulnerabilities.Vulnerabilities.Medium) > 0)
+	case vulnModel.SeverityLow:
+		return prompt &&
+			(len(vulnerabilities.Vulnerabilities.Critical) > 0 ||
+				len(vulnerabilities.Vulnerabilities.High) > 0 ||
+				len(vulnerabilities.Vulnerabilities.Medium) > 0 ||
+				len(vulnerabilities.Vulnerabilities.Low) > 0)
+	}
+
+	return r.Config.GetBool(constants.SecurityPromptConfig)
+}
+
+func (r *RequirementOperation) promptForSecurity(out output.Outputer, vulnerabilities *model.VulnerabilityIngredient) (bool, error) {
+	out.Print(locale.Tr("warning_vulnerable_simple", strconv.Itoa(vulnerabilities.Vulnerabilities.Length())))
+
+	var pkgVersionVulns []string
+	if len(vulnerabilities.Vulnerabilities.Critical) > 0 {
+		criticalOutput := fmt.Sprintf("[RED]%d Critical: [/RESET]", len(vulnerabilities.Vulnerabilities.Critical))
+		criticalOutput += fmt.Sprintf("[CYAN]%s[/RESET]", strings.Join(vulnerabilities.Vulnerabilities.Critical, ", "))
+		pkgVersionVulns = append(pkgVersionVulns, criticalOutput)
+	}
+
+	if len(vulnerabilities.Vulnerabilities.High) > 0 {
+		highOutput := fmt.Sprintf("[ORANGE]%d High: [/RESET]", len(vulnerabilities.Vulnerabilities.High))
+		highOutput += fmt.Sprintf("[CYAN]%s[/RESET]", strings.Join(vulnerabilities.Vulnerabilities.High, ", "))
+		pkgVersionVulns = append(pkgVersionVulns, highOutput)
+	}
+
+	if len(vulnerabilities.Vulnerabilities.Medium) > 0 {
+		mediumOutput := fmt.Sprintf("[YELLOW]%d Medium: [/RESET]", len(vulnerabilities.Vulnerabilities.Medium))
+		mediumOutput += fmt.Sprintf("[CYAN]%s[/RESET]", strings.Join(vulnerabilities.Vulnerabilities.Medium, ", "))
+		pkgVersionVulns = append(pkgVersionVulns, mediumOutput)
+	}
+
+	if len(vulnerabilities.Vulnerabilities.Low) > 0 {
+		lowOutput := fmt.Sprintf("[MAGENTA]%d Low: [/RESET]", len(vulnerabilities.Vulnerabilities.Low))
+		lowOutput += fmt.Sprintf("[CYAN]%s[/RESET]", strings.Join(vulnerabilities.Vulnerabilities.Low, ", "))
+		pkgVersionVulns = append(pkgVersionVulns, lowOutput)
+	}
+
+	out.Print(pkgVersionVulns)
+	out.Print("")
+	out.Print(locale.T("more_info_vulnerabilities"))
+
+	confirm, err := r.Prompt.Confirm("", locale.Tr("prompt_continue_pkg_operation"), ptr.To(false))
```

Comment:
How does this work with non-interactive mode? At first glance, it looks like it will always fail (i.e. `state install vulnerable-package -n` will not work. I could be wrong though. I'm in a few layers too deep, heh.

Reply:
You're right it will not install a package if you have prompting enabled and there are CVEs that are "higher" or equal to your configured preference.

Reply:
So then a follow up question: is this the intended behavior? Should `-n` be a bypass, or should we have a `--force` flag?

Reply:
We discussed this during stand and decided to add an error message with a tip in non-interactive mode. Should cover this use-case.